### ISSUE TITLE
AI does not flag and use shipyards

### ISSUE THREAD

**Issue by kodobi** [vcmi/vcmi#3194]
**Describe the bug**
AI does not flag and/or use shipyards 

**Game logs**
N/A

**To Reproduce**
Start for example Arrogance on 1.4, type cheats for army & map reveal & hide in castle & skip turns

**Expected behavior**
AI will capture shipyard and buy boat within first two weeks

**Actual behavior**
AI will never capture shipyard

**Did it work earlier?**
It works in 1.3.2.

**Screenshots**
N/A

**Version**
 - OS: Android
 - Version:1.4 (from 14th of November I believe)

**Additional context**
N/A

**Comment by JoaoPaul54** [comment]
Just to give some more info. I found that in the PC version 1.3.2 this happens but not always.
Still trying to figure out if it is on a specific castle or map but as far as I see it is kind of random.

**Comment by JoaoPaul54** [comment]
Some extra info after some testing.
It seems to me the issue comes from the creatures defending it as I tried to defeat the guards and then leave. In 3 days (week end in the middle) the AI went to get the shipyard.

So, either the shipyard is considered not worth the fight, or the AI is seeing some creatures as extremely powerful and avoids the fight altogether.

In my test map I saw this happen to 2 of the AI's, one of the shipyards were guarded by sea weeds and the other by dark ones. Both from the abyss mod I believe... might be something related to the mod I guess...
